### ISSUE TITLE
ApplePay is showing even in unsupported browser/devices

### ISSUE THREAD

**Issue by sm131313** [RadarsofthouseDev/billwerk-plus-pay-for-magento2#18]
Magento 2.4.6-p6 
Billwerk V 1.2.46
Aheadworks One page checkout (https://aheadworks.com/one-step-checkout-extension-for-magento-2)

We went from -p5 (everything working fine) to -p6 and now start to see Apple Pay showing up on the frontend in checkout, e.g. in Google Chrome PC or in Android. Did not happen prior to it. 
Apple Pay should only be visible in supported devices/browsers. 

I believe this happened after the latest critical p6 patch that patched the checkout for exploitation.

**Comment by beautycos** [comment]
We see exactly same issue.

**Comment by sm131313** [comment]
Our developer investigated a bit more and found it could also come along with radarsofthouse/reepay upgrade from 1.2.44 to 1.2.46 since we updated it the same time as we patched. 
The extension has removed user agent check that was present before (I don't know, maybe they add something new that doesn't work properly now).
Possibly (guessing only) it is described as "Removed PHP payment validation" https://github.com/RadarsofthouseDev/billwerk-plus-payment-for-magento2/releases/tag/1.2.46)

I think it's better if investigated properly and fixed.

**Comment by beautycos** [comment]
I think it could be due to the update for reepay module, as it should fix a problem where Google Pay doesnt show in checkout. But yes we also patched and updated reepay module today, so it is still also possible it could be the patch.

**Comment by RadarsofthouseDev** [comment]
@sm131313 Could you please send me your website URL to support@radarsofthouse.dk? I just want to check something on Aheadworks one-step-checkout.

**Comment by beautycos** [comment]
@RadarsofthouseDev we use onestepcheckout and have same issue.

**Comment by RadarsofthouseDev** [comment]
@beautycos Noted. I will check onestepcheckout as well.

**Comment by sm131313** [comment]
User-agent check method was removed in the newest version:

file: vendor/radarsofthouse/reepay/Model/Reepayapplepay.php

public function isAvailable(\Magento\Quote\Api\Data\CartInterface $quote = null)
{
    $user_agent = isset($_SERVER['HTTP_USER_AGENT']) ? $_SERVER['HTTP_USER_AGENT'] : '';
    if (stripos($user_agent, 'Edg') !== false) {
        return false;
    } elseif (stripos($user_agent, 'Chrome') !== false) {
        return false;
    } elseif (stripos($user_agent, 'Safari') !== false) {
        return parent::isAvailable($quote);
    } else {
        return false;
    }
}

**Comment by RadarsofthouseDev** [comment]
> User-agent check method was removed in the newest version:
> 
> file: vendor/radarsofthouse/reepay/Model/Reepayapplepay.php
> 
> public function isAvailable(\Magento\Quote\Api\Data\CartInterface $quote = null) { $user_agent = isset($_SERVER['HTTP_USER_AGENT']) ? $_SERVER['HTTP_USER_AGENT'] : ''; if (stripos($user_agent, 'Edg') !== false) { return false; } elseif (stripos($user_agent, 'Chrome') !== false) { return false; } elseif (stripos($user_agent, 'Safari') !== false) { return parent::isAvailable($quote); } else { return false; } }

@sm131313 @beautycos
Yes, we have removed the PHP validation since it is not the same as the validation on the generic payment window. We removed the PHP validation for both Apple Pay and Google Pay. The right solution is that we need to implement the JavaScript validation that Billwerk+ provided.

Reference to Billwerk+ documentation:
- [Apple Pay](https://docu.billwerk.plus/api/en/billwerk-checkout/payment-method-details/apple-pay.html)
- [Google Pay](https://docu.billwerk.plus/api/en/billwerk-checkout/payment-method-details/google-pay.html)

Now, I can make it work with the standard checkout of Magento, but I could not test it with onestepcheckout. I need help from you guys to test the code on Aheadworks one-step-checkout and onestepcheckout.com. I'm making a .patch file and will send it to you guys via email soon.
